### ISSUE TITLE
Ionic CLI 3.7.0 upgrade check only prompts update to 3.8.0, installs 3.8.1 anyway

### ISSUE THREAD

**Issue by janpio** [ionic-team/ionic-cli#2648]
```
Jans-MBP:project sujan$ ionic cordova build android
? The Ionic CLI (local version) has an update available (3.7.0 => 3.8.0)! Would you like to install it? (Y/n
) 
```

But it actually installed 3.8.1 then:

```
cli packages: (/Users/sujan/stromzaehler/node_modules)

    @ionic/cli-utils  : 1.8.1
    ionic (Ionic CLI) : 3.8.1

global packages:

    Cordova CLI : 7.0.1 

local packages:

    @ionic/app-scripts : 2.0.2
    Cordova Platforms  : android 6.2.3 ios 4.4.0
    Ionic Framework    : ionic-angular 3.5.3

System:

    Android SDK Tools : 26.0.2
    ios-deploy        : 1.9.1 
    ios-sim           : 5.0.1 
    Node              : v7.8.0
    npm               : 4.2.0 
    OS                : OS X El Capitan
    Xcode             : Xcode 8.2.1 Build version 8C1002 

```

**Comment by dwieeb** [comment]
This is a pretty rare case... it uses the `latest` dist-tag to install the update, not any specific version.

**Comment by janpio** [comment]
Where does it get the version number of the "current" version from?

**Comment by dwieeb** [comment]
`~/.ionic/daemon.json` - this file is written to by the daemon process, which checks for updates every hour, basically

**Comment by janpio** [comment]
Ok, so I hit a race condition here?

**Comment by dwieeb** [comment]
It could be solved by using the version from `daemon.json` directly, instead of using a dist-tag. You'd just get an older version.

**Comment by janpio** [comment]
Yeah, I am just trying to understand why it even suggested 3.8.0 instead of 3.8.1. Can come in handy next time or in the forum...

**Comment by janpio** [comment]
Same situation now with update from 3.7.0 to 3.8.1 (but installs 3.9.1):

```
PS C:\Users\Jan\Documents> ionic -v
? The Ionic CLI has an update available (3.7.0 => 3.8.1)! Would you like to install it? Yes
> npm install -g ionic@latest
× Running command - failed!
[... unrelevant npm error message - fu windows ...]
PS C:\Users\Jan\Documents> ionic -v
3.9.1
```

**Comment by janpio** [comment]
One more on a Mac:

```
Jans-MBP:~ sujan$ ionic -v
? The Ionic CLI has an update available (3.7.0 => 3.8.0)! Would you like to inst
all it? Yes
> npm install -g ionic@latest
✔ Running command - done!

[OK] Updated Ionic CLI to 3.8.0! 🎉

Please re-run your command.

Note: You can downgrade to your old version by running: npm install -g ionic@3.7.0
Jans-MBP:~ sujan$ ionic -v
3.9.1
Jans-MBP:~ sujan$ 
```
